### DISCUSSION TITLE
Sending mail issue

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#17206] [Archive]
Mails were working OK with a previous version of Directus.

Now that I am on `9.22.4` they are not sent anymore.
Neither the Password lost, nor the mails that were sent by flows.

Not sure it is related to the last version.
My debian server sends correctly mails from the command-line.

How can I test this more deeply through Directus?

**Comment by paescuj** [comment]
I don't think anything related to the mail service has changed in the in the latest versions. Do you see any warning / error messages in the log?

**Comment by Pachat** [comment]
Thank you for your confirmation.
I just restarted the server. And it works now 👍

**Reply by paescuj** [reply]
Great! 🎉

**Comment by Pachat** [comment]
This morning, does not work again. Even after a reboot.
Sending from the command line works, but not from Directus.
mail.log shows the command line sendings but nothing from Directus.
I use Postfix. Are there any particular settings to look at?

Current settings in `.env`
```
### Email

# Email address from which emails are sent ["no-reply@directus.io"]
EMAIL_FROM="app@site.com"

# What to use to send emails. One of
# sendmail, smtp, mailgun, sendgrid, ses.

EMAIL_TRANSPORT="sendmail"
# EMAIL_TRANSPORT="postfix"

EMAIL_SENDMAIL_NEW_LINE="unix"

# EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"
EMAIL_SENDMAIL_PATH="/usr/sbin/postfix"
# EMAIL_SENDMAIL_PATH="/etc/postfix"

## Email (Sendmail Transport)

# What new line style to use in sendmail ["unix"]
# EMAIL_SENDMAIL_NEW_LINE="unix"

# Path to your sendmail executable ["/usr/sbin/sendmail"]
# EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"
EMAIL_SENDMAIL_PATH="/usr/sbin/postfix"

## Email (SMTP Transport)
# EMAIL_SMTP_HOST="localhost"

# Use SMTP pooling
# EMAIL_SMTP_POOL=true
# EMAIL_SMTP_PORT=465
# EMAIL_SMTP_SECURE=false # Use TLS
# EMAIL_SMTP_IGNORE_TLS=false
# EMAIL_SMTP_USER="username"
# EMAIL_SMTP_PASSWORD="password"

## Email (Mailgun Transport)
# EMAIL_MAILGUN_API_KEY="key-1234123412341234"
# EMAIL_MAILGUN_DOMAIN="a domain name from https://app.mailgun.com/app/sending/domains"

## Email (SendGrid Transport)
# EMAIL_SENDGRID_API_KEY="key-1234123412341234"

```

**Reply by paescuj** [reply]
Try with `EMAIL_SENDMAIL_PATH="/usr/sbin/sendmail"` as this is the "sendmail-compatible" binary provided by Postfix.

**Reply by Pachat** [reply]
Tried. But still no mail sent.

**Comment by br41nslug** [comment]
For reliable mail I'd recommend using an external mailing service like Sendgrid or Mailgun. While possible to host your own mailing server it is a complex thing to keep running https://www.digitalocean.com/community/tutorials/why-you-may-not-want-to-run-your-own-mail-server

**Reply by Pachat** [reply]
I agree with that. And that's why my server is set to not receive any mails.

But I would like to at least send Directus mails. Everything is correct: SPF, DKIM, etc.
What's strange is that it worked fine when sending mails with Directus. And suddenly stopped.
I mean, the sendmail executable is no longer accessed at all: nothing in the logs. Except for the mail sent by the command line.
Was it after updating Directus to `9.22.4` or doing an `apt update && apt full-upgrade -y`. Not sure why it suddenly stopped working?

**Reply by paescuj** [reply]
> What's strange is that it worked fine when sending mails with Directus. And suddenly stopped.

For this reason, I think the problem is not related to Directus.

> Was it after updating Directus to 9.22.4 [...]

As already stated in https://github.com/directus/directus/discussions/17206#discussioncomment-4705758 I'm not aware of any changes to the mail service in the last few releases.

**Reply by Pachat** [reply]
OK. I need to do further investigation.

**Comment by Pachat** [comment]
Still cannot send mails from Directus. mail command works properly from the terminal.
I tried sendmail and smtp. Directus does not even hit mail.log
Where in the code are mails sent?

**Comment by Pachat** [comment]
Solved on a clean install (v10)
_As it has been months that I did only updates, may be one of the updates missed the step._
